OrbitCaptureGgpService/main.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitCaptureGgpService.h"
+#include "OrbitVersion/OrbitVersion.h"
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "absl/flags/usage.h"
+#include "absl/flags/usage_config.h"
+
+ABSL_FLAG(uint16_t, grpc_port, 44767, "gRPC server port for capture ggp service");
+ABSL_FLAG(uint16_t, orbit_service_grpc_port, 44765, "gRPC server port for OrbitService");
+ABSL_FLAG(int32_t, pid, 0, "pid to capture");
+ABSL_FLAG(std::vector<std::string>, functions, {},
+          "Comma-separated list of functions to hook to the capture");
+ABSL_FLAG(std::string, file_name, "", "File name used for saving the capture");
+ABSL_FLAG(std::string, file_directory, "/var/game/",
+          "Path to locate orbit file. By default it is /var/game/");
+ABSL_FLAG(std::string, log_directory, "",
+          "Path to locate debug file. By default only stdout is used for logs");
+ABSL_FLAG(uint16_t, sampling_rate, 1000, "Frequency of callstack sampling in samples per second");
+ABSL_FLAG(bool, frame_pointer_unwinding, false, "Use frame pointers for unwinding");
+ABSL_FLAG(bool, thread_state, false, "Collect thread states");
+
+namespace {
+
+std::string GetLogFilePath(const std::string& log_directory) {
+  std::filesystem::path log_directory_path{log_directory};
+  std::filesystem::create_directory(log_directory_path);
+  const std::string log_file_path = log_directory_path / "OrbitClientGgp.log";
+  LOG("Log file: %s", log_file_path);
+  return log_file_path;
+}
+
+}  // namespace
+
+int main(int argc, char** argv) {
+  absl::SetProgramUsageMessage("Orbit CPU Profiler Ggp Client");
+  absl::SetFlagsUsageConfig(absl::FlagsUsageConfig{{}, {}, {}, &OrbitCore::GetBuildReport, {}});
+  absl::ParseCommandLine(argc, argv);
+
+  const std::string log_directory = absl::GetFlag(FLAGS_log_directory);
```

Comment:
If I move it I would not be able to check if the _log_directory_ is empty. If the user does not specify a log directory, it is empty by default and in that case we wouldn't want to call _InitLogFile_ at all as the user would not want to create a log file and it would try to do so. That's why I think it is needed to be here and be provided as an argument instead